File: src/main/java/com/pengrad/telegrambot/model/ChatMember.java
package com.pengrad.telegrambot.model;

/**
 * Stas Parshin
 * 29 May 2016
 */
public class ChatMember {

    public enum Status {
        creator, administrator, member, left, kicked;
    }

    private User user;
    private Status status;

    public User user() {
        return user;
    }

    public Status status() {
        return status;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ChatMember that = (ChatMember) o;

        if (user != null ? !user.equals(that.user) : that.user != null) return false;
        return status == that.status;

    }

    @Override
    public int hashCode() {
        int result = user != null ? user.hashCode() : 0;
        result = 31 * result + (status != null ? status.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "ChatMember{" +
                "user=" + user +
                ", status=" + status +
                '}';
    }
}

File: library/src/main/java/com/pengrad/telegrambot/model/ChatMember.kt
package com.pengrad.telegrambot.model

import java.io.Serializable

/**
 * Stas Parshin
 * 29 May 2016
 */
data class ChatMember(
        @get:JvmName("user") val user: User? = null,
        @get:JvmName("status") val status: Status? = null,
        @get:JvmName("customTitle") val custom_title: String? = null,
        @get:JvmName("untilDate") val until_date: Int? = null,
        @get:JvmName("canBeEdited") val can_be_edited: Boolean? = null,
        @get:JvmName("canPostMessages") val can_post_messages: Boolean? = null,
        @get:JvmName("canEditMessages") val can_edit_messages: Boolean? = null,
        @get:JvmName("canDeleteMessages") val can_delete_messages: Boolean? = null,
        @get:JvmName("canRestrictMembers") val can_restrict_members: Boolean? = null,
        @get:JvmName("canPromoteMembers") val can_promote_members: Boolean? = null,
        @get:JvmName("canChangeInfo") val can_change_info: Boolean? = null,
        @get:JvmName("canInviteUsers") val can_invite_users: Boolean? = null,
        @get:JvmName("canPinMessages") val can_pin_messages: Boolean? = null,
        @get:JvmName("isMember") val is_member: Boolean? = null,
        @get:JvmName("canSendMessages") val can_send_messages: Boolean? = null,
        @get:JvmName("canSendMediaMessages") val can_send_media_messages: Boolean? = null,
        @get:JvmName("canSendPolls") val can_send_polls: Boolean? = null,
        @get:JvmName("canSendOtherMessages") val can_send_other_messages: Boolean? = null,
        @get:JvmName("canAddWebPagePreviews") val can_add_web_page_previews: Boolean? = null
) : Serializable {

    enum class Status {
        creator, administrator, member, restricted, left, kicked
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}